core/src/main/scala/org/http4s/internal/parsing/Rfc2616BasicRules.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.internal.parsing
+
+import cats.parse.{Numbers, Parser, Parser1, Rfc5234}
+
+// Direct implementation of http://www.w3.org/Protocols/rfc2616/rfc2616-sec2.html#sec2
+private[http4s] object Rfc2616BasicRules {
```

Comment:
`cats.parse.Rfc5234` provides almost all of these, and any that aren't should probably go in RFC7320, which supersedes Rfc2616.

I don't know whether it was the right thing to name the parser helpers after the RFCs.  It helps traceability to standards, but maybe not discoverability for developers.

Reply:
Makes sense. I saw the `ListSep` that was being used in the Transfer-Encoding parser was defined here, so I just ported the whole thing.